modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -985,6 +985,95 @@ public void run() {
         StatusThreadPoolExecutor.getInstance().execute(runnableTask);
     }
 
+    @ReactMethod
+    public void multiformatSerializePublicKey(final String multiCodecKey, final String base58btc, final Callback callback) throws JSONException {
+        if (!checkAvailability()) {
+            callback.invoke(false);
+            return;
+        }
+
+        Runnable runnableTask = new Runnable() {
+            @Override
+            public void run() {
+                String res = Statusgo.multiformatSerializePublicKey(multiCodecKey,base58btc);
+                callback.invoke(res);
+            }
+        };
+
+        StatusThreadPoolExecutor.getInstance().execute(runnableTask);
+    }
+
+    @ReactMethod
+    public void multiformatDeserializePublicKey(final String multiCodecKey, final String base58btc, final Callback callback) throws JSONException {
+        if (!checkAvailability()) {
+            callback.invoke(false);
+            return;
+        }
+
+        Runnable runnableTask = new Runnable() {
+            @Override
+            public void run() {
+                String res = Statusgo.multiformatDeserializePublicKey(multiCodecKey,base58btc);
+                callback.invoke(res);
+            }
+        };
+
+        StatusThreadPoolExecutor.getInstance().execute(runnableTask);
+    }
+
+    @ReactMethod
+    public void compressPublicKey(final String multiCodecKey, final Callback callback) throws JSONException {
```

Comment:
Just throwing a comment here to raise awareness, but there's nothing preventing us from better modularizing the file `StatusModule.java` and using it only to expose public methods, but not the actual implementation, which would live in separate classes however we see fit. This separation would further help writing tests for these methods, which are dearly missed.

Currently this file is basically a kitchen sink for random bits of Java code.

Reply:
Hmm, this is worth doing a quick experiment on, I will log my findings here : https://github.com/status-im/status-mobile/issues/14688
Would love to modularize further  🫶🏻